clients/pkg/promtail/client/client.go

```diff
@@ -291,8 +308,14 @@ func (c *client) run() {
 
 			// drop the entry because its length is greater than maxLineSize. maxLineSize == 0 means disabled.
 			if c.maxLineSize != 0 && len(e.Line) > c.maxLineSize {
-				c.metrics.droppedEntries.WithLabelValues(c.cfg.URL.Host, tenantID, DropReasongMaxLineSizeLimited).Inc()
-				break
+				if c.maxLineSizeTruncate {
```

Comment:
Here I would prefer to handle early return case first. So that we don't need to have unwanted `if and else` (reads bit better IMO)

```go
	if !c.maxLineSizeTruncate {
		c.metrics.droppedEntries.WithLabelValues(c.cfg.URL.Host, tenantID, ReasonLineTooLong).Inc()
		c.metrics.droppedBytes.WithLabelValues(c.cfg.URL.Host, tenantID, ReasonLineTooLong).Add(len(e.Line))
		break
	}
	c.metrics.mutatedEntries.WithLabelValues(c.cfg.URL.Host, tenantID, ReasonLineTooLong).Inc()
	c.metrics.mutatedBytes.WithLabelValues(c.cfg.URL.Host, tenantID, ReasonLineTooLong).Add(float64(len(e.Line) - c.maxLineSize))
	e.Line = e.Line[:c.maxLineSize]
```

NOTE: I also update `c.metrics.droppedBytes` when skipping the log line completely.